### PR TITLE
Fix `MissingPresenceValidation` for inclusion/exclusion validators with non Enumerable list of items

### LINKED ISSUE #112: NoMethodError: undefined method `include?' for #<Proc...>
I just installed active_record_doctor 1.10.0 and ran it for the first time and the following error was thrown:

```
% bundle exec rake active_record_doctor --trace
** Invoke active_record_doctor (first_time)
** Invoke active_record_doctor:setup (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute active_record_doctor:setup
** Execute active_record_doctor
rake aborted!
NoMethodError: undefined method `include?' for #<Proc:0x0000000113da8568 /Users/manuel/code/myapp/app/app/models/admin_users/admin_user.rb:46 (lambda)>

            !validator.options.fetch(:in, []).include?(nil)
                                             ^^^^^^^^^
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:57:in `block in inclusion_validator_present?'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:54:in `any?'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:54:in `inclusion_validator_present?'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:46:in `validator_present?'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:31:in `block (2 levels) in detect'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:29:in `each'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:29:in `block in detect'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:26:in `each'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/missing_presence_validation.rb:26:in `detect'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/base.rb:50:in `run'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/detectors/base.rb:17:in `run'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/runner.rb:15:in `block in run_one'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/errors.rb:5:in `handle_exception'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/runner.rb:14:in `run_one'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/runner.rb:26:in `block in run_all'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/runner.rb:25:in `each'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/runner.rb:25:in `run_all'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.10.0/lib/active_record_doctor/rake/task.rb:60:in `block in define'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/sentry-ruby-5.4.2/lib/sentry/rake.rb:26:in `execute'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:83:in `block in run'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:80:in `run'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/manuel/.rbenv/versions/3.1.2/bin/rake:25:in `load'
/Users/manuel/.rbenv/versions/3.1.2/bin/rake:25:in `<top (required)>'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/cli/exec.rb:58:in `load'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/cli/exec.rb:58:in `kernel_load'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/cli/exec.rb:23:in `run'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/cli.rb:485:in `exec'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/cli.rb:31:in `dispatch'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/cli.rb:25:in `start'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/exe/bundle:48:in `block in <top (required)>'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'
/Users/manuel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.20/exe/bundle:36:in `<top (required)>'
/Users/manuel/.rbenv/versions/3.1.2/bin/bundle:25:in `load'
/Users/manuel/.rbenv/versions/3.1.2/bin/bundle:25:in `<main>'
Tasks: TOP => active_record_doctor
```

The code at admin_user.rb:46 is:

```
validates :my_option, inclusion: { in:      -> { _1.role?(:superadmin, :admin) ?
                                                 [true, false] :
                                                 [false] },
                                   message: "can only be selected for admins with role superadmin or admin" }
```

Let me know if I can do anything else to debug!

### PR DESCRIPTION
**Pull request by fatkodima**
Fixes #112.

### CODE REVIEW

File: lib/active_record_doctor/detectors/missing_presence_validation.rb [hunk 1]
```diff
@@ -79,6 +79,11 @@ def presence_validator_present?(model, column)
             (validator.attributes & allowed_attributes).present?
         end
       end
+
+      def inclusion_validator_items(validator)
+        items = validator.options[:in] || validator.options[:within] || []
+        items.is_a?(Enumerable) ? items : []
```
Comment (gregnavis) [inline lib/active_record_doctor/detectors/missing_presence_validation.rb:85]: It seems that a `Proc` validator is equivalent to `in: []`. I don't think that's a good choice for users, as `active_record_doctor` shouldn't be reporting errors indiscriminately.

For now, I suggest _skipping_ those validators that use a `Proc`.

**Review by gregnavis — CHANGES_REQUESTED** [review]

**Comment by fatkodima** [comment]
@gregnavis Applied the changes.

File: test/active_record_doctor/detectors/missing_presence_validation_test.rb [hunk 2]
```diff
@@ -91,6 +91,16 @@ def test_non_null_boolean_is_not_reported_if_nil_excluded
     refute_problems
   end
 
+  def test_non_null_boolean_is_reported_if_exclusion_is_proc
```
Comment (gregnavis) [inline test/active_record_doctor/detectors/missing_presence_validation_test.rb:94]: @fatkodima, please fix the test case name (I think it should say `..._is_not_reported_...`) and it's good to merge.

**Review by gregnavis — APPROVED** [review]

File: test/active_record_doctor/detectors/missing_presence_validation_test.rb (hunk 2 above)
Comment (fatkodima) [reply test/active_record_doctor/detectors/missing_presence_validation_test.rb]: @gregnavis Oops. Fixed.

File: test/active_record_doctor/detectors/missing_presence_validation_test.rb [hunk 3]
```diff
@@ -91,6 +91,16 @@ def test_non_null_boolean_is_not_reported_if_nil_excluded
...
 
+  def test_non_null_boolean_is_not_reported_if_exclusion_is_proc
+    create_table(:users) do |t|
+      t.boolean :active, null: false
+    end.create_model do
+      validates :active, exclusion: { in: ->(_user) { [nil] } }
+    end
+
+    refute_problems
+  end
```
Comment (gregnavis) [inline test/active_record_doctor/detectors/missing_presence_validation_test.rb:102]: @fatkodima, could you add a similar test case for `inclusion`?

**Review by gregnavis — APPROVED** [review]

**Pull request merged**